tests/cpp/test_resharding.cpp

```diff
@@ -82,97 +82,98 @@ TEST_F(ReshardingTest, Detection) {
 
   TensorView* tv0 = makeContigTensor(3);
   fusion->addInput(tv0);
+  TensorView* tv1 = set(tv0);
+  TensorView* tv2 = set(tv1); // resharding
```

Comment:
Why moving all those lines at the top? The problem is that the comment "resharding"/"not resharding" are based on the device mesh, which is now set far for the line

Reply:
I think it might be the ordering issue Jingyue brought up in the meeting (short example in the meeting notes)

Reply:
Yes, I mentioned that in the description as well. But Sam's concern is still valid that the comments and the shardings will be too far from each other. I'm not quite happy with that either. 

I can make some changes to the test: 
1. still shard immediately after the op (and therefore the comment), but the annotation will be based on the already propagated shardings. E.g. [reset](https://github.com/NVIDIA/Fuser/pull/2744/files#diff-c5175dc597d3e4dd05ea38630935938b69c33328284a9e3b689a60ccaaef5b7bR204-R205) a parallel type to `Serial` for all gather. 
2. Split the test so we have only one op per test. This is more typing but follows the best testing practice to keep tests independent.

Reply:
I merged the PR with ReshardingTest.Detection split into one op per test.